### ISSUE TITLE
Use 'Throw' expression causes erroneous code

### ISSUE THREAD

**Issue by znakeeye** [dotnet/roslyn#19507]
**Version Used**: 15.2 (26430.4)

**Steps to Reproduce**:

```csharp
public Foo(T bar)
{
	if (bar == null)  // Use 'Throw' expression here...
	{
		throw new ArgumentNullException(nameof(bar));
	}
}
```
	
**Error CS0019 Operator '??' cannot be applied to operands of type 'T' and '\<throw expression>'**

**However, if I add a constraint on T in my class it works:**

```csharp
public sealed class Foo<TClass> where TClass : class
{}
```

**Expected Behavior**:
The "Use Throw expression" command should not be available or it should complain about the type. Maybe suggest adding the "where"-constraint above?

**Actual Behavior**:
Compilation error CS0019.

**Comment by sharwell** [comment]
It appears our analyzer is ignoring the rule described in #17754.

**Comment by Pilchie** [comment]
Already fixed for 15.3 as part of https://github.com/dotnet/roslyn/pull/18178.
